### ISSUE TITLE
Error Install Cartoview Geonode 2.5.15 + Ubuntu 14.04

### ISSUE THREAD

**Issue by dpadron2** [cartologic/cartoview#13]
Hello,

I try to install Cartoview in Ubuntu 14.04 + Geonode 2.5.15+ Django 1.8.17

An error is encountered while executing: python manage.py makemigrations app_manager

Any ideas?

root@ide:/home/geonode/geonode/idesca/idesca# bower install --allow-root
bower ol3#^3.18.2               cached https://github.com/baasgeo/ol3-bower.git#3.18.2
bower ol3#^3.18.2             validate 3.18.2 against https://github.com/baasgeo/ol3-bower.git#^3.18.2
bower angular#1.5.9             cached https://github.com/angular/bower-angular.git#1.5.9
bower angular#1.5.9           validate 1.5.9 against https://github.com/angular/bower-angular.git#1.5.9
bower angular-messages#1.5.9    cached https://github.com/angular/bower-angular-messages.git#1.5.9
bower angular-messages#1.5.9  validate 1.5.9 against https://github.com/angular/bower-angular-messages.git#1.5.9
bower angular-aria#1.5.9        cached https://github.com/angular/bower-angular-aria.git#1.5.9
bower angular-aria#1.5.9      validate 1.5.9 against https://github.com/angular/bower-angular-aria.git#1.5.9
bower angular-animate#1.5.9     cached https://github.com/angular/bower-angular-animate.git#1.5.9
bower angular-animate#1.5.9   validate 1.5.9 against https://github.com/angular/bower-angular-animate.git#1.5.9
bower angular-material#^1.1.1   cached https://github.com/angular/bower-material.git#1.1.3
bower angular-material#^1.1.1 validate 1.1.3 against https://github.com/angular/bower-material.git#^1.1.1
bower angular-resource#1.5.9    cached https://github.com/angular/bower-angular-resource.git#1.5.9
bower angular-resource#1.5.9  validate 1.5.9 against https://github.com/angular/bower-angular-resource.git#1.5.9
bower angular-resource-tastypie#^1.0.4           cached https://github.com/mw-ferretti/angular-resource-tastypie.git#1.0.4
bower angular-resource-tastypie#^1.0.4         validate 1.0.4 against https://github.com/mw-ferretti/angular-resource-tastypie.git#^1.0.4
bower angular-bootstrap#^2.4.0                   cached https://github.com/angular-ui/bootstrap-bower.git#2.5.0
bower angular-bootstrap#^2.4.0                 validate 2.5.0 against https://github.com/angular-ui/bootstrap-bower.git#^2.4.0
bower angular-drag-and-drop-lists#^2.1.0         cached https://github.com/marceljuenemann/angular-drag-and-drop-lists.git#2.1.0
bower angular-drag-and-drop-lists#^2.1.0       validate 2.1.0 against https://github.com/marceljuenemann/angular-drag-and-drop-lists.git#^2.1.0
bower ng-image-appear#^1.11.3                    cached https://github.com/ArunMichaelDsouza/ng-image-appear.git#1.11.3
bower ng-image-appear#^1.11.3                  validate 1.11.3 against https://github.com/ArunMichaelDsouza/ng-image-appear.git#^1.11.3
bower lf-ng-md-file-input#^1.5.1                 cached https://github.com/shuyu/angular-material-fileinput.git#1.5.1
bower lf-ng-md-file-input#^1.5.1               validate 1.5.1 against https://github.com/shuyu/angular-material-fileinput.git#^1.5.1
bower normalize-css#^5.0.0                       cached https://github.com/necolas/normalize.css.git#5.0.0
bower normalize-css#^5.0.0                     validate 5.0.0 against https://github.com/necolas/normalize.css.git#^5.0.0
bower ol3#^3.18.2                               install ol3#3.18.2
bower angular-messages#1.5.9                    install angular-messages#1.5.9
bower angular-material#^1.1.1                   install angular-material#1.1.3
bower angular-aria#1.5.9                        install angular-aria#1.5.9
bower angular#1.5.9                             install angular#1.5.9
bower angular-animate#1.5.9                     install angular-animate#1.5.9
bower angular-drag-and-drop-lists#^2.1.0        install angular-drag-and-drop-lists#2.1.0
bower angular-bootstrap#^2.4.0                  install angular-bootstrap#2.5.0
bower angular-resource-tastypie#^1.0.4          install angular-resource-tastypie#1.0.4
bower ng-image-appear#^1.11.3                   install ng-image-appear#1.11.3
bower angular-resource#1.5.9                    install angular-resource#1.5.9
bower normalize-css#^5.0.0                      install normalize-css#5.0.0
bower lf-ng-md-file-input#^1.5.1                install lf-ng-md-file-input#1.5.1

ol3#3.18.2 static/vendor/ol3

angular-messages#1.5.9 static/vendor/angular-messages
âââ angular#1.5.9

angular-material#1.1.3 static/vendor/angular-material
âââ angular#1.5.9
âââ angular-animate#1.5.9
âââ angular-aria#1.5.9
âââ angular-messages#1.5.9

angular-aria#1.5.9 static/vendor/angular-aria
âââ angular#1.5.9

angular#1.5.9 static/vendor/angular

angular-animate#1.5.9 static/vendor/angular-animate
âââ angular#1.5.9

angular-drag-and-drop-lists#2.1.0 static/vendor/angular-drag-and-drop-lists

angular-bootstrap#2.5.0 static/vendor/angular-bootstrap
âââ angular#1.5.9

angular-resource-tastypie#1.0.4 static/vendor/angular-resource-tastypie
âââ angular#1.5.9
âââ angular-resource#1.5.9

ng-image-appear#1.11.3 static/vendor/ng-image-appear

angular-resource#1.5.9 static/vendor/angular-resource
âââ angular#1.5.9

normalize-css#5.0.0 static/vendor/normalize-css

lf-ng-md-file-input#1.5.1 static/vendor/lf-ng-md-file-input
âââ angular-material#1.1.3

-------------------------------
root@ide:/home/geonode/geonode/idesca# **python manage.py makemigrations app_manager**
Traceback (most recent call last):
  File "manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 354, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python2.7/dist-packages/django/core/management/__init__.py", line 328, in execute
    django.setup()
  File "/usr/local/lib/python2.7/dist-packages/django/__init__.py", line 18, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/usr/local/lib/python2.7/dist-packages/django/apps/registry.py", line 85, in populate
    app_config = AppConfig.create(entry)
  File "/usr/local/lib/python2.7/dist-packages/django/apps/config.py", line 86, in create
    module = import_module(entry)
  File "/usr/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: No module named geonode-client

**Comment by hishamkaram** [comment]
Hello @dpadron2 
Solution one (better for this case):
this error because `django-geonode-client` not installed.
you can install it using` pip install django-geonode-client` after this retry ```python manage.py makemigrations app_manager```.
Solution two : 
upgrade cartoview using this command 
`pip install cartoview -U`
and follow steps in docs

**Comment by dpadron2** [comment]
Thank you very much for your answer, Hishamkaram

Now it's working at http://204.12.225.85:8000

Now I need help installing the Apps

- Execute correctly "python manage.py loaddata json / default_oauth_apps.json"

- Now having doubt should I replace the local_setting of geonode in the project cartoview?
I do not understand this, you can help me.

- I can not install the Apps http://204.12.225.85:8000/apps/manage/

Thanks for your help

**Comment by hishamkaram** [comment]
Hello @dpadron2 
I am Happy to hear from you.
I can see that you configured cartoview correctly But geoserver not configured well .
So apps won't work correctly.
- First stepto do before installing apps 
     - this [link](http://docs.geonode.org/en/master/tutorials/admin/geoserver_geonode_security/#geoserver-security-backend) is a good starting point to configure geoserver

- second step you need to install apps manually because you are you using old version of cartoview that unfortunately contain bugs with app_manager Don't worry I will show you how to install apps:
     1. visit github and choose the you want to install download it
     2. extract it and rename extracted folder to the same name with ```master```
     3. in cartoview project folder you will find apps folder copy the app folder in the previos step to this folder then create file with this name ```apps.yml``` and add the following lines

            
                  - name: <app_name>
                    active: true
                    order: 0
            
         replace ```<app_name>``` with your app name
     4. open admin panel from this [link](http://204.12.225.85:8000/admin/app_manager/app/) click add new app button on upper right corner fill information as follow:
     
          ![capture](https://cloud.githubusercontent.com/assets/24391550/23278597/c24f553c-fa1a-11e6-8a26-3d47f213e1f6.PNG)
          ![capture1](https://cloud.githubusercontent.com/assets/24391550/23278598/c2594510-fa1a-11e6-809b-b7e078fda94d.PNG)
          ![capture2](https://cloud.githubusercontent.com/assets/24391550/23278599/c25a2502-fa1a-11e6-8bde-c22f0ecd3505.PNG)
    5. click save button 
    6. go to apps tab in your website you can see the app and use it

**or**
Follow this [guide](http://cartologic.github.io/cartoview/#install-on-ubuntu-linux) to install the new version without any errors**

**Comment by dpadron2** [comment]
Hello Hisham,

I have reinstalled everything, I have NOT even installed cartoview.

Following your advice, I configured Geoserver, but I have a problem that I can not solve.

I installed Geonode 2.5.15 + Ubuntu 14.4 + Geoserver-2.9.x-oauth2

Uploading a layer generates the following error:

Tried to make a GET request to http://localhost:8080/geoserver/rest/workspaces/default.xml but got a 404 status code: No such workspace: default

http://204.12.225.85/

User: demo
Password: demo123

Anyone have any idea of the origin of this error?

Thanks for your attention.


Diego Padron
